### PR TITLE
Update linter/formatter and apply new logging principles.

### PR DESCRIPTION
**Pull request by samuelhwilliams**
### Change description
This moves fsd-utils away from flake8+black to `ruff`, which post-award has been using with success (as in: no-one cares or misses flake8+black).

Ruff makes it easy to include some extra linting rules around logging to cover some basics. We also want to apply a new principle to our log messages: don't log arbitrary python objects.

_In general_, this PR prefers the %-style string formatting, but that's not a hard-and-fast or enforced rule. What is enforced is that strings in logging can't be eagerly-formatted. I prefer %-style in this case because the `{}` style format strings have historically caused some issues with our fsd-utils autoformatting for log messages (albeit this is mostly when including python dicts in the log message, which themselves have `{}` characters). I don't have a strong enforceable opinion on which style to use, but % is a fine default.

### CODE REVIEW

File: fsd_utils/mapping/application/qa_mapping.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def extract_questions_and_answers(forms, language=EN) -> dict:
                             else:
                                 questions_answers[form_name][field["title"]] = answer
     except Exception as e:
-        current_app.logger.error(f"Error occurred while processing form data: {e}")
-        current_app.logger.error(f"Could not map the data for form: {form_name}")
-        raise Exception(f"Could not map the data for form: {form_name}")
+        current_app.logger.error("Error occurred while processing form data: %s", e)
+        current_app.logger.error("Could not map the data for form: %s", form_name)
+        raise Exception(f"Could not map the data for form: {form_name}") from e
```
Comment (albertkol) [inline fsd_utils/mapping/application/qa_mapping.py:84]: Missed one.

File: fsd_utils/sqs_scheduler/task_executer_service.py [hunk 2]
```diff
@@ -79,20 +79,22 @@ def _handle_message_receiving_and_processing(self):
                 self.visibility_time,
                 self.sqs_wait_time,
             )
-            self.logger.debug(f"{thread_id} Message Count [{len(sqs_messages)}]")
+            self.logger.debug("%s Message Count [%S]"), thread_id, {len(sqs_messages)}
```
Comment (albertkol) [inline fsd_utils/sqs_scheduler/task_executer_service.py:82]: Is there a difference between `%s` and `%S`? 
My test throws an error with capital S.

**Review by albertkol — APPROVED** [review]
Aye

File: fsd_utils/sqs_scheduler/task_executer_service.py (hunk 2 above)
Comment (samuelhwilliams) [reply fsd_utils/sqs_scheduler/task_executer_service.py:82]: also good spot and probably be just fat-fingering 🥲

File: fsd_utils/mapping/application/qa_mapping.py (hunk 1 above)
Comment (samuelhwilliams) [reply fsd_utils/mapping/application/qa_mapping.py:84]: wait what did i miss?

**Review by albertkol — APPROVED** [review]
.

**Pull request merged**